src/components/material/MaterialReviews.tsx

```diff
@@ -1,30 +1,63 @@
 import React from "react";
-import {
-  ExternalReview,
-  InfomediaReview,
-  LibrariansReview
-} from "../../core/dbc-gateway/generated/graphql";
+import { useGetReviewManifestationsQuery } from "../../core/dbc-gateway/generated/graphql";
+import { Pid } from "../../core/utils/types/ids";
 import ReviewExternal from "./ReviewExternal";
 import ReviewInfomedia from "./ReviewInfomedia";
 import ReviewLibrarian from "./ReviewLibrarian";
+import ReviewSkeleton from "./ReviewSkeleton";
 
 export interface MaterialReviewsProps {
-  listOfReviews: Array<LibrariansReview | ExternalReview | InfomediaReview>;
+  pids: Pid[];
+  dataCy?: string;
 }
 
 export const MaterialReviews: React.FC<MaterialReviewsProps> = ({
-  listOfReviews
+  pids,
+  dataCy = "material-reviews"
 }) => {
+  const { data, isLoading } = useGetReviewManifestationsQuery({
+    pid: pids
+  });
+
+  if (isLoading) {
+    return <ReviewSkeleton />;
+  }
+
+  if (!data) {
+    return null;
+  }
+
+  const { manifestations: reviews } = data;
+
   return (
-    <ul className="reviews">
-      {listOfReviews.map((review) => {
-        if (review.__typename === "ExternalReview") {
-          return <ReviewExternal review={review} />;
+    <ul className="reviews" data-cy={dataCy}>
+      {reviews.map((review) => {
+        if (
```

Comment:
I think it would be pettier with a function returning the review type with the same precedense as you have here. Then you could use a switch case outside the render function to determine which component to return.

Reply:
Mmmm I must say I don't think it's prettier, but I did it, and it's commited.